js/plates-model/model-worker.js

```diff
@@ -80,13 +85,15 @@ onmessage = function modelWorkerMsgHandler (event) {
     if (clickedField) {
       plateDrawTool(clickedField.plate, clickedField.id, data.type === 'drawContinent' ? 'continent' : 'ocean')
     }
+  } else if (data.type === 'takeSnapshot') {
+    takeSnapshot()
   } else if (data.type === 'restoreSnapshot') {
     let serializedModel
     if (snapshots.length === 0) {
       serializedModel = initialSnapshot
     } else {
       serializedModel = snapshots.pop()
-      if (snapshots.length > 0 && model.stepIdx < serializedModel.stepIdx + 20) {
+      if (snapshots.length > 0 && serializedModel.stepIdx > 0 && model.stepIdx < serializedModel.stepIdx + 20) {
```

Comment:
Also, we could leave a comment there that this new condition is important for authoring case, so it doesn't get removed in the future.